scripts/mempool.js

```diff
@@ -1,381 +1,375 @@
      */
-    removeFromState(cNewUTXO, nState) {
-        const arrPendingUTXOs = this.getUTXOsByState(nState);
-        // Loop each pending UTXO
-        for (const cUTXO of arrPendingUTXOs) {
-            // Search for matching ID + output number
-            if (cUTXO.id === cNewUTXO.id && cUTXO.vout === cNewUTXO.vout) {
-                // Nuke it from orbit
-                this.removeUTXO(cUTXO);
-                break;
-            }
-        }
-    }
-
+    #balance = 0;
     /**
-     * Add a new UTXO to the wallet
-     * @param {UTXO} UTXO
+     * @type {number} - Our Cold Staking balance in Satoshis
      */
-    addUTXO({
-        id,
-        path,
-        sats,
-        script,
-        vin,
-        vout,
-        height,
-        status,
-        isDelegate,
-        isReward,
-    }) {
-        const newUTXO = new UTXO({
-            id,
-            path,
-            sats,
-            script,
-            vin,
-            vout,
-            height,
-            status,
-            isDelegate,
-            isReward,
-        });
-
-        if (this.isAlreadyStored({ id, vout })) {
-            this.updateUTXO({ id, vout });
-        } else {
-            // If this new UTXO is a reward with one vin (i.e: a Stake), we'll backtrace the input
-            if (isReward && vin?.length === 1) {
-                const cStakeInput = this.getUTXO(vin[0].txid, vin[0].vout);
-
-                // And if the input is in our wallet, remove it
-                if (cStakeInput) {
-                    this.removeUTXO(cStakeInput);
-                }
-            }
-            this.UTXOs.push(newUTXO);
-        }
-
-        // Re-render the Balance UIs
-        getBalance(true);
-        getStakingBalance(true);
-    }
-
+    #coldBalance = 0;
     /**
-     * Update an existing UTXO, by confirming its pending status
-     * The UTXO must be in
-     * @param {Object} UTXO - Object to be deconstructed
-     * @param {String} UTXO.id - Transaction id
-     * @param {Number} UTXO.vout - vout
+     * @type {Number}
+     * The maximum block height that we received with the call 'utxos'
+     * We don't want to receive anymore transactions which are below this block
      */
-    updateUTXO({ id, vout }) {
-        if (debug) {
-            console.assert(
-                this.isAlreadyStored({ id, vout }),
-                'Debug Mode: updateUTXO must be called with an existing UTXO'
-            );
-        }
-        const cUTXO = this.UTXOs.find(
-            (utxo) => utxo.id === id && utxo.vout == vout
-        );
-        switch (cUTXO.status) {
-            case Mempool.PENDING:
-                cUTXO.status = Mempool.CONFIRMED;
-                break;
-        }
-        getBalance(true);
-        getStakingBalance(true);
+    #syncHeight = -1;
+    constructor() {
+        /**
+         * Multimap txid -> spent Coutpoint
+         * @type {Multimap<String, COutpoint>}
+         */
+        this.spent = new Multimap();
```

Comment:
I think we could still remove this `Multimap` dependency with a simple in-house version, a dependency for this single variable is too heavy imo.

Reply:
as discussed on discord I will create our Multimap in a second PR